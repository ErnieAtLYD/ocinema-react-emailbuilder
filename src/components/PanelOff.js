// @flow

import React, { useEffect, useRef, useState } from 'react';
import Button from '@material-ui/core/Button';
import InputLabel from '@material-ui/core/InputLabel';
import Typography from '@material-ui/core/Typography';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import ExportTemplate from './ExportTemplate'
import { useFetchOPosts } from './useFetch'

type PanelType = {
  createLayoutItem: any => void,
  exportAsHTML: any => void,
  layout: string
};

const PanelOff = ({ createLayoutItem, exportAsHTML, layout }: PanelType) => {
  const inputLabel = useRef(null);
  const [labelWidth, setLabelWidth] = useState(0);
  const [values, setValues] = useState("");

  useEffect(() => {
    const refWidth = inputLabel.current ? inputLabel.current.offsetWidth : 0;
    setLabelWidth(refWidth);
  }, []);

  const response = useFetchOPosts();

  const handleChange = event => {
    setValues(event.target.value);
  };

  const handleImport = () => {
    if (!values) return;

    let searchedfilm = response.events.filter(film => film.id === values);
    searchedfilm = searchedfilm[0];

    let temp = {
      id: searchedfilm.id,
      layout: 'filmlayout',
      content: searchedfilm.title,
      htmldescription: searchedfilm.description,
      htmlquotes: searchedfilm.event_reviews,
      posterurl: searchedfilm.image.sizes.medium.url,
      showtimes: searchedfilm.event_showtimes,
      agileurl: searchedfilm.url,
      bannerurl: ''
    }
    createLayoutItem(temp);
  };

  return (
    <>
      <Button
        variant="contained"
        onClick={() => createLayoutItem()}>
        Add a blank section
      </Button>

      <Typography>Import from the website:</Typography>
      <InputLabel ref={inputLabel} htmlFor="event-label">Event</InputLabel>
      <Select
        value={values}
        onChange={handleChange}
        labelWidth={labelWidth}
        inputProps={{
          name: 'age',
          id: 'event-label',
        }}>
        <MenuItem value=""><em>Choose an event</em></MenuItem>
        {response.events && response.events.map( (film) => (
            <MenuItem key={film.id} value={film.id}>
              {film.title}
            </MenuItem>
          )
        )}
      </Select>
      <Button
        variant="contained"
        disabled={!response || values === ''}
        onClick={handleImport}
      >
        Import
      </Button>
      <ExportTemplate
        exportAsHTML={exportAsHTML}
        layout={layout}
      />
    </>
  )
}

export default PanelOff;
