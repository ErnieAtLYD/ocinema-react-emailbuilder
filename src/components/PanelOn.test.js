import React from "react";
import { Button } from "@material-ui/core";
import { createShallow } from "@material-ui/core/test-utils";
import PanelOff from "./PanelOff";

let shallow;
let wrapper;

// see: https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-1-162ce7466128

beforeEach(() => {
  shallow = createShallow();
  wrapper = shallow(<PanelOff />);
});

describe("PanelOff", () => {
  it("renders without crashing", () => {
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });

  it("should call the onClick function when 'Add a blank section' button is clicked", () => {
    // const mockedFn = jest.fn();
    // wrapper.instance().createLayoutItem = mockedFn;
    // console.log(wrapper.debug());
    // wrapper.find(Button).first().props().onClick();
    // expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
