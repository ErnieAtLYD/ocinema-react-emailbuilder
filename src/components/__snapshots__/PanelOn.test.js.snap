// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelOff renders without crashing 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="contained"
  >
    Add a blank section
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Typography))>
    Import from the website:
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(InputLabel))
    htmlFor="event-label"
  >
    Event
  </WithStyles(ForwardRef(InputLabel))>
  <WithStyles(ForwardRef(Select))
    inputProps={
      Object {
        "id": "event-label",
        "name": "age",
      }
    }
    labelWidth={0}
    onChange={[Function]}
    value=""
  >
    <WithStyles(ForwardRef(MenuItem))
      value=""
    >
      <em>
        Choose an event
      </em>
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
  <WithStyles(ForwardRef(Button))
    disabled={true}
    onClick={[Function]}
    variant="contained"
  >
    Import
  </WithStyles(ForwardRef(Button))>
  <ExportTemplate />
</Fragment>
`;
